ada_2_fhir-r4/zibs2024/MAPPING_GUIDELINES.md

```diff
@@ -0,0 +1,78 @@
+* `concat($urlBaseNictizProfile, $profileName[English zib name])` in all other cases.
+
+
+### Zibs resulting in multiple FHIR resources
+Sometimes, the transformation of a single ADA instance can result in multiple FHIR resources. For example, zib HeartRate has a focal Observation profile that represents the zib itself, and two associated Observation profiles for the concepts HeartbeatRegularity and InterpretationHeartRate (which are referenced from the focal Observation).
+
+As described above, a profile template is created for each resource that needs to be created, with name and mode set to the profile id and a match on the corresponding ADA element. The template for the focal profile normally acts as the entry point for the transformation. So in addition to producing the focal resource, it is also responsible for kicking off the transformations to the associated resources. The result of the transformation is thus a _list_ of resources.
+
+### Zibs having multiple translations
+A zib can also have different meanings depending on the context. For example, zib Procedure can represent a planned or a performed procedure, resulting in either a ServiceRequest or a Procedure resource.
+
+In this case there again is a profile template for each resource that needs to be created (but all are matching on the same ADA instance). This means that the choice of the template to use is usually made by the caller of the template.
+
+If a choice as to which transformation can be made based on the content of the ADA instance, a wrapper template can be defined to call the proper template. It usually will have a name and mode set to `nl-core-[zib name]` (which is available, as the different versions of the profiles will have some postfix attached to their name) and a match on the ADA element being transformed.
+
+### Multiple zibs resulting in single FHIR resource
+In some rare situations, multiple zibs end up in the same FHIR resource. For example zib Nationality, which will usually be mapped to an element in the Patient resource.
+
+In this situation, there is no stylesheet of profile template for the "guest" zibs. Instead, the ADA instance for which the result should end up in the focal resource is defined as a parameter to template of the focal resource. This means that it is up to the caller of the template to preprocess the ADA input in such a way that the profile is called in with the proper parameter.
+
+## The _generateDisplay template(s)
+
+For each profile template, there must be a template with mode set to `_generateDisplay` and a match on the ADA instance element. This template is used to generate a display for references to the resulting instance.
+
+## Branching strategy
+
+For each new stylesheet, a branch is made from master. When the profile is added to the package, the new work is merged with master.
```

Comment:
Or additions/changes to an already existing/published stylesheet? E.g. in zib2020 we added some "guest" zibs of the Patient later on within the same template; this might occur in zib2024 as well with new zibs

Reply:
Good point. This sentence reasons from "what to do when adding a new profile mapping", but it applies to changes as well.

Reply:
Rewrote this section